### PR TITLE
Update tree.rst

### PR DESCRIPTION
**Pull request by MarwanSalim**
Improve the documentation so developers know how to use it for custom `parent_id`. Related issues: https://github.com/cakephp/cakephp/issues/11657 and https://github.com/cakephp/cakephp/pull/11662

### CODE REVIEW

**Comment by othercorey** [comment]
Instead of duplicating examples, I would create note section with a link to the threaded finder.

```
.. note::
    For more information on 'threaded' finder options see <link>.
```

File: en/orm/behaviors/tree.rst [hunk 1]
```diff
@@ -95,6 +95,16 @@ in a hierarchy, you can stack the 'threaded' finder::
         echo "{$child->name} has " . count($child->children) . " direct children";
     }
 
+While, if you’re using custom ``parent_id`` you need to pass it in the ‘threaded’ finder option (i.e. ``parentField``) . Read more on the   
+`Finding Threaded Data logic <https://book.cakephp.org/4/en/orm/retrieving-data-and-resultsets.html#finding-threaded-data>`_::
```
Comment (markstory) [inline en/orm/behaviors/tree.rst]: We could use a `:ref:finding-threaded-data` link instead. This link won't work properly in PDF and ePub outputs.

**Review by Arhell — APPROVED** [review]

File: en/orm/behaviors/tree.rst (hunk 1 above)
Comment (MarwanSalim) [reply en/orm/behaviors/tree.rst]: @othercorey @markstory 
b263aaf
Thank you

File: en/orm/behaviors/tree.rst [hunk 2]
```diff
@@ -95,6 +95,11 @@ in a hierarchy, you can stack the 'threaded' finder::
         echo "{$child->name} has " . count($child->children) . " direct children";
     }
 
+While, if you’re using custom ``parent_id`` you need to pass it in the 'threaded' finder option (i.e. ``parentField``) . 
+
+.. note::
+    For more information on 'threaded' finder options see :ref:`Finding Threaded Data logic <finding-threaded-data>`
```
Comment (othercorey) [inline en/orm/behaviors/tree.rst:101]: The `finding-threaded-data` ref doesn't exist. Need to add it above header in https://github.com/cakephp/docs/blob/4.x/en/orm/retrieving-data-and-resultsets.rst

```
.. _finding-threaded-data:

Finding Threaded Data
=====================
```

Comment (MarwanSalim) [reply en/orm/behaviors/tree.rst:101]: https://github.com/cakephp/docs/pull/7422#issue-1238439633
Thank you

**Comment by markstory** [comment]
Thank you 🎉

**Pull request merged**